### DISCUSSION TITLE
Always get latest git dependency

### DISCUSSION THREAD

**Discussion by zotille** [npm/feedback#92] [CLI] [answered]
Our tram have a depency A used by multi projects. So we add the repository of A as dependency in other prokects.
While A is update frequently, and we wanna always get latest commit of A, but the version would be locked by package-lock.json like below:
![image](https://user-images.githubusercontent.com/38178024/98837603-1dd66080-247e-11eb-98e7-9a42917dca99.png)

Can anyone tell me how to get always get latest code?

**Comment by ljharb** [comment]
The point of a lockfile is to always get the same code. If you disable or remove the lockfile, I’d expect the code to be updated every time - alternatively, you can `npm update test_type`?

**Reply by zotille** [reply]
Emmmm.. I've tried add install script in package.json ,
![image](https://user-images.githubusercontent.com/38178024/98839083-144df800-2480-11eb-92b2-8fcea4a6c803.png)

Then it works, but the install seems to do twice.

**Comment by ethomson** [comment, marked as answer]
Hello @zotille!

Thanks for the question, and I'm sorry if you're having difficulty with npm.  This is the repo for product feedback and helps guide the product team on decision making and prioritization of new feature work.  You may want to check [the product documentation](https://docs.npmjs.com/), and if that doesn't help, [Stack Overflow](https://stackoverflow.com/) is a great place to ask questions.

Thanks for using npm!

**Reply by zotille** [reply]
Sorry for this topic.
but finally, can you consider add a config ignore dependencies from lock file .

**Reply by MylesBorins** [reply]
@zotille so the specific feature you are requesting is to be able to have a lockfile for most dependencies, but to ignore it for other dependencies? I'm struggling a bit to think about how that could be done from a UX perspective.

One thing that you could attempt would be to turn on dependabot, it can actually keep git dependencies up to date!

https://dependabot.com/javascript/
